### PR TITLE
feat: add referer-restriction plugin form

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.
image

**Related issues**

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by membphis** [comment]
Can you provide a screenshot of the effect?

**Comment by LiteSun** [comment]
> Can you provide a screenshot of the effect?

sure.

File: web/cypress/support/commands.js [hunk 1]
```diff
@@ -66,6 +66,16 @@ Cypress.Commands.add('configurePlugins', (cases) => {
...
+
+        cy.get(domSelector.codeMirrorMode).invoke('text').then(text => {
+          if (text === 'Form') {
+            cy.get(domSelector.codeMirrorMode).should('exist');
+            cy.get(domSelector.codeMirrorMode).click();
+            cy.get(domSelector.selectDropdown).should('be.visible');
+            cy.get(domSelector.selectJSON).click();
+          }
+        });
+
```
Comment (jbampton) [inline web/cypress/support/commands.js:78]:
```suggestion

```

File: web/src/components/Plugin/UI/referer-restriction.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        valuePropName="checked"
+        tooltip='Whether to bypass the check when the Referer header is missing or malformed.'
+      >
+        <Switch />
+      </Form.Item>
+    </Form>
+  );
+}
+
+export default RefererRestriction;
```
Comment (jbampton) [inline web/src/components/Plugin/UI/referer-restriction.tsx:109]:
```suggestion
export default RefererRestriction;

```

Comment (LiteSun) [reply web/src/components/Plugin/UI/referer-restriction.tsx]: updated, tks.

**Review by juzhiyuan — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Comment by juzhiyuan** [comment]
image

@LiteSun Your new changes failed when building

File: web/cypress/support/commands.js [hunk 3]
```diff
@@ -66,6 +66,18 @@ Cypress.Commands.add('configurePlugins', (cases) => {
 
         // NOTE: wait for the Drawer to appear on the DOM
         cy.focused(domSelector.drawer).should('exist');
+
+        cy.get(domSelector.codeMirrorMode).invoke('text').then(text => {
+          if (text === 'Form') {
+            // FIXME: https://github.com/cypress-io/cypress/issues/7306
```
Comment (juzhiyuan) [inline web/cypress/support/commands.js:72]: Do we have issue for this?

**Review by juzhiyuan — APPROVED** [review]
Just rechecked with https://github.com/apache/apisix/blob/master/apisix/plugins/referer-restriction.lua, please update i18n in the new PR.

File: web/cypress/support/commands.js (hunk 3 above)
Comment (LiteSun) [reply web/cypress/support/commands.js:72]: sure. #1738

**Review by moonming — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Pull request merged**